tpi/terraform.py

```diff
@@ -143,3 +143,14 @@ def _shell_default(
         port = f":{port}" if port is not None else ""
         cmd.append(f"{user}{host}{port}")
         run(cmd)
+
+    def state_mv(self, source, destination, **kwargs):
+        """Retain an existing remote object but track it as a
+        different resource instance address.
+        """
+        assert source and destination
+
+        name = source.split(".")[-1]
+
+        with self.make_tf(name) as tf:
+            tf.cmd("state mv", f"{source}", f"{destination}")
```

Comment:
aren't these expected to be strings anyway?

```suggestion
            tf.cmd("state mv", source, destination)
```